Update SonarQube step config doc: View scan results in SonarQube portal

tejakummarikuntla opened the pull request:
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://669804d23d3ccf0deae218bb--harness-developer.netlify.app

douglas-j-bothwell commented on docs/security-testing-orchestration/sto-techref-category/sonarqube-sonar-scanner-reference.md:
```diff
@@ -319,6 +319,36 @@ If there's a proxy between your Harness pipeline and your SonarQube server, you
+
+
+
+1. Set the “Analysis Type” to "manual" under the “Scan Tool” section. This brings up a new field called "Branch Name".
+2. In the "Branch Name" field, you can define the desired name for the scan result as it should appear in the SonarQube portal.
+
```
```suggestion
1. Under **Scan Tool**, set the **Analysis Type** to **Manual**. 
2. Set the **Branch Name** to the name you want to use for your scan results in the SonarQube portal.
```

I think you can tighten this up. Often you don't need to say "the ___ section."  Here's a good discussion from the Microsoft style guide: 
https://learn.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui

Also, don't use the word "desire" -- use "want" instead.

https://developers.google.com/style/word-list#desire

douglas-j-bothwell commented on docs/security-testing-orchestration/sto-techref-category/sonarqube-sonar-scanner-reference.md:
```diff
@@ -319,6 +319,36 @@ If there's a proxy between your Harness pipeline and your SonarQube server, you
+
+### Best practices
+
+You can utilize [Harness expressions](https://developer.harness.io/docs/platform/variables-and-expressions/harness-variables/) to automatically populate values from triggers. For example, you can define the "Branch Name" as `<+trigger.sourceBranch>/<+trigger.prNumber>`. This will generate a naming convention in the format SOURCE_BRANCH_NAME/PR_NUMBER, which is particularly useful when triggering  ...
```
```suggestion
You can use [Harness expressions](https://developer.harness.io/docs/platform/variables-and-expressions/harness-variables/) to automatically populate values from triggers. For example, you can define the "Branch Name" as `<+trigger.sourceBranch>/<+trigger.prNumber>`. This will generate a naming convention in the format SOURCE_BRANCH_NAME/PR_NUMBER, which is particularly useful when triggering scans on a specific PR. 
```

I think you mean "use" here, not "utilize"....
https://developers.google.com/style/word-list#utilize

douglas-j-bothwell reviewed (approved):
Approved, assuming you accept my feedback....

tejakummarikuntla commented:
> Approved, assuming you accept my feedback....

Thank you, @douglas-j-bothwell, this is very helpful. I've addressed the reviews; also, thanks for the references.

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://6698e0f20599fb8300ca6e2c--harness-developer.netlify.app
